package com.jcirmodelsquad.tcjcir.vehicles.locomotives.diesel;

import net.minecraft.entity.item.EntityMinecart;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.nbt.NBTTagList;
import net.minecraft.world.World;
import net.minecraftforge.common.util.Constants;
import train.common.Traincraft;
import train.common.api.DieselTrain;
import train.common.api.LiquidManager;
import train.common.core.util.TraincraftUtil;
import train.common.library.EnumSounds;
import train.common.library.EnumTrains;
import train.common.library.GuiIDs;
import train.common.library.sounds.SoundRecord;

public class DieselSW8 extends DieselTrain {
    @Override
    public SoundRecord getSoundRecord()
    {
        return EnumSounds.DieselSW8;
    }
    public DieselSW8(World world) {
        super(world, EnumTrains.SW8.getTankCapacity(), LiquidManager.dieselFilter());
        
        InsertTexture(0, "DRIR 553");
        InsertTexture(1, "Blandsville & Blankerston");
        InsertTexture(2, "NYC");
        InsertTexture(3, "NYC");
        InsertTexture(4, "Strasbooger/Strasboner/Strasburger/Strasburry Dough Boy");//if you change this i will end your life
        InsertTexture(5, "EMDX");
    }


    

    @Override
    public void updateRiderPosition() { TraincraftUtil.updateRider(this, -0.08, 0.2, -0.35); }



    



    

    @Override
    public float getOptimalDistance(EntityMinecart cart) { return 1.1F;
    }

    

    @Override
    public String getInventoryName() {
        return "EMD SW8";
    }

    




}
